src/network/networkd-network.c

```diff
@@ -1575,6 +1575,48 @@ int config_parse_section_route_table(
         return 0;
 }
 
+int config_parse_dhcp_max_client_attempt(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Network *network = data;
+        uint64_t k;
+        int r;
+
+        assert(network);
+        assert(lvalue);
+        assert(rvalue);
+
+        if (isempty(rvalue)) {
+                network->dhcp_max_client_attempt = 0;
+                return 0;
+        }
+
+        if (streq(rvalue, "infinity"))
+                network->dhcp_max_client_attempt = (uint64_t) -1;
+        else {
+                r = config_parse_uint64(unit, filename, line, section, section_line, lvalue, ltype, rvalue, &k, userdata);
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r,
+                                   "Failed to parse DHCP maximum client attempt: %s, ignoring.", rvalue);
+                        return 0;
+                }
+
+                network->dhcp_max_client_attempt = k;
```

Comment:
The temporary variable is not needed, it's OK to just call `config_parse_uint64(..., &network->dhcp_max_client_attempt, ...)` directly.